src/test/util_threadnames_tests.cpp

```diff
@@ -26,15 +26,17 @@ const std::string TEST_THREAD_NAME_BASE = "test_thread.";
  *
  * @return the set of name each thread has after attempted renaming.
  */
+
+static Mutex lock;
+static std::set<std::string> names GUARDED_BY(lock);
```

Comment:
In commit "test/util_threadnames_tests: add thread safety annotations" (d53072ec730d8eec5a5b72f7e65a54b141e62b19)

Assuming the reason these need to be global is lock annotations not working on local variables (didn't check), better IMO to simply not to use lock annotations here than to introduce global variables and make it unsafe to call RenameEnMasse from different threads itself. 

Other option would be to make RenameEnMasse a member function in a class with Mutex and std::set members so it could work non-globally and also be annotated

Reply:
It's not unsafe to call RenameEnMasse from different threads with the global lock, just inefficient. I tried the class approach but it seemed like overkill, so I've dropped the commit instead.